JL.Core/Mining/MiningUtils.cs

```diff
@@ -440,6 +440,77 @@ public static async Task MineToFile(LookupResult lookupResult, string currentTex
         Utils.Logger.Information("Mined {PrimarySpelling}", lookupResult.PrimarySpelling);
     }
 
+    public static async Task<bool> CheckDuplicate(LookupResult lookupResult)
+    {
+        CoreConfigManager coreConfigManager = CoreConfigManager.Instance;
+        if (!coreConfigManager.AnkiIntegration || !coreConfigManager.CheckForDuplicateCards)
+        {
+            return false;
+        }
+
+        Dictionary<MineType, AnkiConfig>? ankiConfigDict = await AnkiConfig.ReadAnkiConfig().ConfigureAwait(false);
+        if (ankiConfigDict is null)
+        {
+            return false;
+        }
+
+        AnkiConfig? ankiConfig;
+        if (DictUtils.s_wordDictTypes.Contains(lookupResult.Dict.Type))
+        {
+            ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Word);
+        }
+        else if (DictUtils.s_kanjiDictTypes.Contains(lookupResult.Dict.Type))
+        {
+            ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Kanji);
+        }
+        else if (DictUtils.s_nameDictTypes.Contains(lookupResult.Dict.Type))
+        {
+            ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Name);
+        }
+        else
+        {
+            ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Other);
+        }
+
+        if (ankiConfig is null)
+        {
+            return false;
+        }
+
+        Dictionary<string, JLField> userFields = ankiConfig.Fields;
+        Dictionary<JLField, string> miningParams = new()
```

Comment:
We probably should use `GetMiningParameters` here. The implementation provided here just partially creates mining parameters but for all we know the user might have `SourceText` as their note type's first field (which is all Anki cares for when checking whether if a card is a duplicate AFAIK) and its value is missing here.

Or better yet, we can check what's the first `Field` in the config and just create a note with that field.